processing/src/main/java/io/druid/query/DefaultQueryMetrics.java

```diff
@@ -45,18 +45,17 @@ public DefaultQueryMetrics(ObjectMapper jsonMapper)
   @Override
   public void query(QueryType query)
   {
-    dataSource(query);
+    dataSources(query);
     queryType(query);
-    interval(query);
     hasFilters(query);
     duration(query);
     queryId(query);
   }
 
   @Override
-  public void dataSource(QueryType query)
+  public void dataSources(QueryType query)
   {
-    builder.setDimension(DruidMetrics.DATASOURCE, DataSourceUtil.getMetricName(query.getDataSource()));
+    builder.setDimension(DruidMetrics.DATASOURCE, DataSourceUtil.getMetricName(query.getDataSources()));
```

Comment:
Should use multi-value dimension for queries with multiple data sources, see `setDimension(String, String[])` method